extensions/guacamole-auth-restrict/src/main/resources/controllers/timeRestrictionFieldController.js

```diff
@@ -0,0 +1,222 @@
+        var restrArray = restrString.split(";");
+
+        // Loop through split string and process each item
+        for (let i = 0; i < restrArray.length; i++) {
+            if (restrictionRegex.test(restrArray[i])) {
+                var currArray = restrArray[i].match(restrictionRegex);
+                var restrDays = currArray[1].split("-");
+                
+                if (restrDays.length > 1) {
+                    for (let j = restrDays[0]; j <= restrDays[1]; j++) {
+                        var entry = {};
+                        entry.weekDay = parseInt(j);
+                        entry.startTime = new Date(1970, 1, 1, parseInt(currArray[2].slice(0,2)), parseInt(currArray[2].slice(2)), 0, 0);
+                        entry.endTime = new Date(1970, 1, 1, parseInt(currArray[3].slice(0,2)), parseInt(currArray[3].slice(2)), 0, 0);
+                        restrictions.push(entry);
+                    }
+                }
+                else {
+                    var entry = {};
+                    entry.weekDay = parseInt(restrDays[0]);
+                    entry.startTime = new Date(1970, 1, 1, parseInt(currArray[2].slice(0,2)), parseInt(currArray[2].slice(2)), 0, 0);
+                    entry.endTime = new Date(1970, 1, 1, parseInt(currArray[3].slice(0,2)), parseInt(currArray[3].slice(2)), 0, 0);
+                    restrictions.push(entry);
+                }
+            }
+        }
+        
+        return restrictions;
+        
+    };
+    
+    /**
+     * Parse the restrictions in the field into a string that can be stored
+     * in an underlying module.
+     * 
+     * @param {Object[]} restrictions
+     *     The array of restrictions that will be converted to a string.
+     * 
+     * @returns {String}
+     *     The string containing the restriction data that can be stored in e.g.
+     *     a database.
+     */
+    var storeRestrictions = function storeRestrictions(restrictions) {
+        // If there are no members of the array, just return an empty string.
+        if (restrictions === null || restrictions.length < 1)
+            return '';
+        
+        var restrString = '';
+        for (let i = 0; i < restrictions.length; i++) {
+            // If any of the properties are not defined, skip this one.
+            if (!Object.hasOwn(restrictions[i], 'weekDay')
+                    || restrictions[i].weekDay === null
+                    || !Object.hasOwn(restrictions[i], 'startTime')
+                    || restrictions[i].startTime === null 
+                    || !Object.hasOwn(restrictions[i], 'endTime') 
+                    || restrictions[i].endTime === null)
+                continue;
+            
+            // If this is not the first item, then add a semi-colon separator
+            if (restrString.length > 0)
+                restrString += ';';
+            var currString = '' + restrictions[i].weekDay;
```

Comment:
This is a pretty dense code block - some newlines between these `if` statements (and some comments about what they're doing) would make this a lot more readable.

Reply:
Spread it out and added some comments.